packages/superset-ui-core/src/SupersetClient.js

```diff
@@ -0,0 +1,180 @@
+import callApi from './callApi';
+
+const AUTH_QUEUE_POLL_MS = 30;
+
+class SupersetClient {
+  constructor(config) {
+    const {
+      protocol = 'http',
+      host = '',
+      headers = {},
+      mode = 'same-origin',
+      timeout,
+      credentials,
+    } = config;
+
+    this.headers = headers;
+    this.host = host;
+    this.mode = mode;
+    this.timeout = timeout;
+    this.protocol = protocol;
+    this.credentials = credentials;
+    this.csrfToken = null;
+    this.didAuthSuccessfully = false;
+    this.requestingCsrf = false;
+
+    this.getUnauthorizedError = this.getUnauthorizedError.bind(this);
+    this.waitForCSRF = this.waitForCSRF.bind(this);
+  }
+
+  isAuthenticated() {
+    return this.didAuthSuccessfully;
+  }
+
+  init() {
+    return this.getCSRFToken();
+  }
+
+  getCSRFToken() {
+    this.requestingCsrf = true;
+
+    // If we can request this resource successfully, it means that the user has
+    // authenticated. If not we throw an error prompting to authenticate.
+    return callApi({
+      credentials: this.credentials,
+      headers: {
+        ...this.headers,
+      },
+      method: 'GET',
+      mode: this.mode,
+      timeout: this.timeout,
+      url: this.getUrl({ endpoint: 'superset/csrf_token/', host: this.host }),
+    })
+      .then(response => {
```

Comment:
Perhaps use `.then(onFulfilled, onRejected)`
instead of `.then(onFulfilled).catch(onRejected)`

`.catch(onRejected)` is equivalent to`.then(undefined, onRejected)`

The current `catch` clause will handle `Promise.reject` thrown from the `then` clause as well, which has the same result in this case, but is redundant work.

Reply:
sure, I think the `.catch` is a little more readable but don't have a super strong opinion.

Reply:
done